src/index.js

```diff
@@ -0,0 +1,311 @@
+   2. Add/update TOWER_ACCESS_TOKEN with your Seqera Platform token
+   3. Ensure your workflow uses: access_token: \${{ secrets.TOWER_ACCESS_TOKEN }}`);
+    }
+    
+    if (!inputs.pipeline) {
+      throw new Error('pipeline is required');
+    }
+    
+    // Initialize API client
+    const apiClient = new SeqeraPlatformAPI({
+      baseUrl: inputs.apiEndpoint,
+      accessToken: inputs.accessToken,
+      debug: inputs.debug
+    });
+    
+    // Test API connectivity
+    core.info('🔗 Testing API connectivity...');
+    logMessage(logFile, 'Testing API connectivity...');
+    const connectionTest = await apiClient.testConnection();
+    if (!connectionTest.success) {
+      logMessage(logFile, `API connectivity test failed: ${connectionTest.error}`, 'ERROR');
+      throw new Error(`API connectivity test failed: ${connectionTest.error}`);
+    }
+    core.info('✅ API connectivity confirmed');
+    logMessage(logFile, 'API connectivity confirmed');
+    
+    // Launch the workflow
+    core.info('🎯 Launching workflow...');
+    logMessage(logFile, 'Launching workflow...');
+    const launchResult = await apiClient.launchWorkflow(inputs);
+    
+    if (!launchResult.success) {
+      // Provide detailed error information
+      let errorMessage = `Workflow launch failed: ${launchResult.error}`;
+      
+      if (launchResult.statusCode === 401) {
+        errorMessage += '\n\n💡 This usually indicates an invalid or expired access token.';
+        errorMessage += '\n   Please check that your TOWER_ACCESS_TOKEN secret is valid.';
+      } else if (launchResult.statusCode === 403) {
+        errorMessage += '\n\n💡 This usually indicates insufficient permissions.';
+        errorMessage += '\n   Please check workspace permissions and compute environment access.';
+      } else if (launchResult.statusCode === 404) {
+        errorMessage += '\n\n💡 This usually indicates the pipeline or workspace was not found.';
+        errorMessage += '\n   Please check the pipeline URL and workspace ID.';
+      }
+      
+      if (launchResult.details && inputs.debug) {
+        errorMessage += `\n\n🐛 Debug details: ${launchResult.details}`;
+      }
+      
+      throw new Error(errorMessage);
+    }
+    
+    const workflowData = launchResult.data;
+    core.info(`✅ Workflow launched successfully!`);
+    core.info(`📊 Workflow ID: ${workflowData.workflowId}`);
+    logMessage(logFile, `Workflow launched successfully! ID: ${workflowData.workflowId}`);
+    
+    // Build workflow URL (this might need adjustment based on actual API responses)
+    let workflowUrl = workflowData.workflowUrl;
+    if (!workflowUrl) {
+      // Construct URL if not provided by API
+      const baseUrl = inputs.apiEndpoint.replace('/api.', '/').replace('/api', '');
+      if (inputs.workspaceId) {
+        workflowUrl = `${baseUrl}/orgs/-/workspaces/${inputs.workspaceId}/watch/${workflowData.workflowId}`;
```

Comment:
Is that `/-/` legit? Looks wrong? I think the org name needs to go there:

https://cloud.seqera.io/orgs/nf-core/workspaces/AWSmegatests/watch/5JOXGt08UAFuw9/v2/logs

Also it's the workspace _name_ and not the workspace _ID_

Reply:
I think that this URL is used only to print to the log for the user to easily click and load the run in the browser. So we want the web URL (not API). There it follows the URL structure I mentioned above, I assume that `GET` parameters don't work.